ktor-client/ktor-client-plugins/ktor-client-auth/common/test/io/ktor/client/plugins/auth/AuthTokenHolderTest.kt

```diff
@@ -63,4 +63,68 @@ class AuthTokenHolderTest {
         assertNotNull(first.await())
         assertNotNull(second.await())
     }
+
+    @Test
+    @OptIn(DelicateCoroutinesApi::class)
+    fun testClearCalledWhileLoadingTokens() = testSuspend {
+        val monitor = Job()
+
+        var clearTokenCalled = false
+        val holder = AuthTokenHolder {
+
+            // suspend until clearToken is called
+            while (!clearTokenCalled) {
+                delay(10)
+            }
+
+            monitor.join()
+            BearerTokens("1", "2")
+        }
+
+        val first = GlobalScope.async(Dispatchers.Unconfined) {
+            holder.loadToken()
+        }
+
+        val second = GlobalScope.async(Dispatchers.Unconfined) {
+            holder.clearToken()
+            clearTokenCalled = true
+        }
+
+        monitor.complete()
+        assertNotNull(first.await())
+        assertNotNull(second.await())
+        assertTrue(clearTokenCalled)
+    }
+
+    @Test
+    @OptIn(DelicateCoroutinesApi::class)
+    fun testClearCalledWhileSettingTokens() = testSuspend {
+        val monitor = Job()
+
+        var clearTokenCalled = false
+        val holder = AuthTokenHolder<BearerTokens> {
+            fail("unexpected call")
```

Comment:
The message here can be more clear: loadTokens argument function shouldn't be invoked